### PR TITLE
DMSG reconnection tests

### PR DESCRIPTION
**Pull request by nkryuchkov**
Moved from https://github.com/evanlinjin/skywire/pull/8.

The tests have some space for improvements that will be done in https://github.com/skycoin/skywire/issues/420. For example, there're some code snippets duplicated among the tests that may be done in separate functions. Due to the requirement of modification of all tests, such code is left as it is. It would be appreciated if improvement proposals for reconnection tests only would be left here. For common improvement proposals, please leave comments in the related issue.

### CODE REVIEW

File: pkg/dmsg/server_test.go [hunk 1]
```diff
@@ -767,6 +765,115 @@ func TestServer_Serve(t *testing.T) {
...
+		})
+
+		t.Run("Random address", func(t *testing.T) {
+			t.Parallel()
+			testReconnect(t, true)
+		})
+	})
+}
+
+func testReconnect(t *testing.T, randomAddr bool) {
```
Comment (evanlinjin) [inline pkg/dmsg/server_test.go:906]: Looks good, however some comments describing what is being tested will be good. Also in regards to naming convention; initiator should go with responder, and local should go with remote.

**Review by evanlinjin — APPROVED** [review]

**Review by Darkren — APPROVED** [review]
Looks good to me

**Comment by ivcosla** [comment]
By the travis logs it looks like the newly added tests are producing a lot of log messages. Logging should be turned off before merging this branch.

At the moment there is a conflict that needs to be fixed too.

File: pkg/dmsg/server_test.go (hunk 1 above)
Comment (nkryuchkov) [reply pkg/dmsg/server_test.go:906]: @evanlinjin Thanks for your review. I'll do this in https://github.com/skycoin/skywire/issues/420

**Pull request merged**